File: Languages/PHP/Day008.php
<!-- Languages - PHP - Day 8 -->
<!-- Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables. -->

<!-- Create a program that takes a "user's" first and last name as input, concatenates them into a full name, slices the full name to extract specific characters, and formats the result in a greeting message.-->

<?php
function greeting($first, $last) {
    $full_name = $first . " " . $last;
    $first_initial = strtoupper(substr($first, 0, 1));
    $last_initial = strtoupper(substr($last, 0, 1));
    return "Hello, " . $full_name . "! Your initials are " . $first_initial . "." . $last_initial . ".";
}

echo "Please enter your first name: ";
$first_name = trim(fgets(STDIN));

if (empty($first_name)) {
    echo "First name cannot be empty.\n";
    exit;
}

echo "Please enter your last name: ";
$last_name = trim(fgets(STDIN));

if (empty($last_name)) {
    echo "Last name cannot be empty.\n";
    exit;
}

echo greeting($first_name, $last_name) . "\n";
?>

File: Languages/C/Day008.c
// Languages - C - Day 8
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a program that takes a "user's" first and last name as input, concatenates them into a full name, slices the full name to extract specific characters, and formats the result in a greeting message.

#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main()
{
    char firstName[20];
    char lastName[20];
    char fullName[40];
    char greeting[100];

    printf("Enter your first name: ");
    scanf("%19s", firstName);

    if (strlen(firstName) == 0)
    {
        printf("First name cannot be empty.\n");
        return 1;
    }

    printf("Enter your last name: ");
    scanf("%19s", lastName);

    if (strlen(lastName) == 0)
    {
        printf("Last name cannot be empty.\n");
        return 1;
    }

    strcpy(fullName, firstName);
    strcat(fullName, " ");
    strcat(fullName, lastName);

    char firstInitial = toupper(firstName[0]);
    char lastInitial = toupper(lastName[0]);

    sprintf(greeting, "Hello, %s! Your initials are %c.%c.", fullName, firstInitial, lastInitial);

    printf("%s\n", greeting);

    return 0;
}
